### ISSUE TITLE
Paths rendering issue

### ISSUE THREAD

**Issue by Nohus** [p-lr/MapCompose#85]
When using the Paths feature, the paths are drawn with artifacts / empty spaces at each connecting point:

![image](https://github.com/p-lr/MapCompose/assets/11064768/a4c340fa-5483-41ea-b9b4-f91c6597e4c5)

For comparison, the same path drawn with the Google Maps Compose library. Notice how the line is fully solid throughout:

![image](https://github.com/p-lr/MapCompose/assets/11064768/ef5e4f85-5112-4500-96e5-af01c8712288)

**Comment by p-lr** [comment]
This due to how path are drawn. MapCompose uses `Canvas.drawLines` which is hardware accelerated, but rendering isn't as pretty as `Canvas.drawPath` as your image shows. 
Historically, I chose to ditch `drawPath` because the performance difference can clearly be felt when rendering long paths. I was ok with it since in this case I value performance more than aesthetics.
Now a few years later I gave `drawPath` another try. It's in the very experimental `drawpath` branch. 

While the demo performs well, my testing on a real app shows a noticeable performance difference. Also, I've seen paths blinks under some circumstances (but this is probably unrelated to `drawPath`).

It would be nice to investigate this performance issue and see if there's a way to use `drawPath` while not hindering performance. In the end, I agree paths could look nicer. However I will be careful about performance.

**Comment by p-lr** [comment]
@Nohus Could you test the branch `drawpath` and give me your feedback? It should render path like Google Maps Compose library, plus with a path decimation algorithm to fix performance on zoom out.

**Comment by Nohus** [comment]
Impressive work! I can confirm it works well.

I noticed a small issue: the paths seem to be drawn without anti-aliasing and have jagged edges, especially visible at particular angles. I tried setting `antiAliasing = true` on the `Paint` object but didn't seem to help.

The path simplification is really good, but I think it's a little bit too aggressive, I can see my paths jumping around as I zoom in and out. I think it may work better for you (and be more required) because you are using paths made of a lot of points. My points are no very dense already.

Updating the epsilon argument from `2.dp` to `1.dp` made it more sensible, at least on my path data.
```
val epsilon = with(LocalDensity.current) {
        (1.dp / zoomPRState.scale).toPx().toDouble()
    }
```
